collaboration/extending-kyma-cli.md

```diff
@@ -0,0 +1,346 @@
+  * Api name to expose with
+  * replicas
+  * min/max memory
+  * min/max cpu
+  * runtime
+  * debug command (to be used with telepresence. This can help if we want to have different runtimes)
+* <function-code.extension> where the developer can write his logic to test. The `extension` is language specific.
+* `dependecies.extension` file with list of dependencies. eg. `package.json` for node and `go.mod and go.sum` for golang dependencies
+
+>The templates generated should be placed a directory in the kyma cli. We should have templates for each language that can be supported. In the future if we are supporting more runtime we can move it to separate repo.
+
+##### config.yaml
+
+This file basically enables developers to configure their functions. With init command this file is created and it basically has some default values. The readme files provide information about various options present in this file
+
+Developer can modify this file to deploy or update a function on to the kyma cluster. This would enable the developer not only tune the function (by modifying replica, memory cpu), but also provide an overview of the current triggers for the function.
+
+Usually one does not change the config so often during development compared to code. So our aim should be to make deployment and testing the function simpler. So having a config file taking care of configuration would make deployment much simpler.
+
+Having such a config file would help developers with limited or no kubernetes knowledge to get themselves aquainted with various configuration options for function and concentrate on the business logic for writing the functions.
+
+Additionally it can be used automated deployments of functions. As this file can be read by kyma cli to deloy function with required configurations.
+
+Example structure of config.yaml
+
+```yaml
+name: <function_name>
+namespace: <namespace> #default by default
+events:
+  - event1
+  - event2
+environment-vars:
+  - foo:bar
+api:
+  url: <url>
+  actions:
+    - GET
+    - POST
+  security:
+    - JWT
+replicas: 1
+memory:
+  min:
+  max:
+cpu:
+  min:
+  max:
+runtime:
+debug-string:
+```
+
+##### Template files
+
+###### <function-code.extension>
+
+  Out of the box from the init command we should give an easy example which can be extended by the developer later. The example should be possible to be deployed directly on the kyma cluster. Here below there is a node example
+
+  ```js
+    module.exports = {
+      main: function (event, context) {
+        return 'Hello World';
+      }
+    };
+  ```
+
+###### <dependecies.extension>
+
+We should generate an empty package.json file which can be edited by developer as and when needed.
+
+###### index.js
+
+We should provide `index.js`(which would start the server and serve the handler function) out of the box which can be used by the developer to test his code locally.
+
+###### readme
+
+We should also provide a readme.md which would consist of following:
```

Comment:
You can mention "instructions" here and delete it from the following list items.

Reply:
fixed.